metrics/tests/named_metric.rs

```diff
@@ -7,9 +7,24 @@ use rustcommon_metrics::*;
 #[metric(name = "custom-name")]
 static METRIC: Counter = Counter::new();
 
+#[metric(
+    name = "custom-name-with-description",
+    description = "some metric with a description"
+)]
+static METRIC_WITH_DESCRIPTION: Counter = Counter::new();
+
 #[test]
 fn metric_name_as_expected() {
     let metrics = metrics().static_metrics();
-    assert_eq!(metrics.len(), 1);
-    assert_eq!(metrics[0].name(), "custom-name");
+    assert_eq!(metrics.len(), 2);
+    assert_eq!(metrics[1].name(), "custom-name");
+    assert_eq!(metrics[1].description(), "");
```

Comment:
I think I'd rather have this be None. It does force the caller to use an `unwrap_or()` or do some other handling, but I think that makes it easier for the caller to be aware that descriptions may not be present and force an explicit decision on how to handle that case.